Makefile

```diff
@@ -170,6 +170,19 @@ deps\:mocks:
 deps\:playground:
 	cd $(PLAYGROUND_DIRECTORY) && npm install --legacy-peer-deps && npm run build
 
+/PHONY: deps\:embedding
+deps\:embedding:
+ifeq ($(OS_GENERAL),Linux)
+	curl -fsSL https://ollama.com/install.sh | sh
+else
+# run ollama in the background
+	nohup ollama serve > ollama.log 2>&1 &
+endif
+# make sure ollama is running before continuing
+	time curl --retry 5 --retry-connrefused --retry-delay 0 -sf http://localhost:11434
+# pull a model
+	ollama pull nomic-embed-text
```

Comment:
suggestion: I prefer the name to be `deps:ollama` instead of `deps:embedding`.

suggestion: Installing ollama, running ollama and pulling a model should be a separate command.
Perhaps:
- `make deps:ollama` to install ollama.
- `make ollama` to serve/start in background. 
- `make ollama:nomic` to pull the model

suggestion: If it is not Linux, can just print that the user must install it manually if have to use UI to download.

Note: the linux vs non-linux differences can be handled in the individual `ifeq/else` blocks in the separate make rules suggested above if you do want to provide them.

Reply:
Done